### PR TITLE
index update/relevance details

### PR DESCRIPTION
**Pull request by kkraune**
more details, based on musings from @bratseth and @toregge - anyone or @geirst please review/merge

tried to separate better index structure in aspects of updates and relevance using separate sections

### CODE REVIEW

File: documentation/proton.html [hunk 1]
```diff
@@ -123,41 +123,76 @@ <h3 id="sub-databases">Sub-databases</h3>
...
+
+
+<h3 id="updates">Updates</h3>
+<p>
+For all <a href="reference/search-definitions-reference.html#index">indexed fields</a>,
+proton has a memory index for the recent changes, implemented using a B-tree.
+This is periodically <a href="#memory-index-flush">flushed</a> to a disk-based posting list index.
+Disk-based indexes are subsequently <a href="#disk-index-fusion">merged</a>.
+</p><p>
+This means that recent changes are in-memory B-tree changes,
```
Comment (geirst) [inline documentation/proton.html]: I don't understand the semantics of this sentence. Consider rewrite: "Updating the in-memory B-trees is lock-free, implemented using copy-on-write semantics". That the changes are recent is already mentioned in the previous paragraph.

File: documentation/proton.html [hunk 2]
```diff
@@ -123,41 +123,76 @@ <h3 id="sub-databases">Sub-databases</h3>
...
-Index fields are string fields and are used for text search.
+Index fields are string fields, used for text search.
 Other field types are <a href="attributes.html">attributes</a>.
+</p>
+
+
+<h3 id="updates">Updates</h3>
+<p>
+For all <a href="reference/search-definitions-reference.html#index">indexed fields</a>,
+proton has a memory index for the recent changes, implemented using a B-tree.
```
[the remark this answers is not in the document]
Comment (kkraune) [reply documentation/proton.html]: ok - fixed

File: documentation/proton.html [hunk 3]
```diff
@@ -123,41 +123,76 @@ <h3 id="sub-databases">Sub-databases</h3>
...
+This is periodically <a href="#memory-index-flush">flushed</a> to a disk-based posting list index.
+Disk-based indexes are subsequently <a href="#disk-index-fusion">merged</a>.
+</p><p>
+This means that recent changes are in-memory B-tree changes,
+implemented with copy-on-write semantics - updating these structures is lock-free.
+This gives high performance, with a predictable steady-state CPU/memory use.
+The driver for this design is the requirement for a sustained, high change rate,
+with stable, predictable read latencies and small temporary increases in CPU/memory.
+</p><p>
```
[the remark this answers is not in the document]
Comment (kkraune) [reply documentation/proton.html]: fixed

File: documentation/proton.html (hunk 1 above)
Comment (kkraune) [reply documentation/proton.html]: agree

**Pull request merged**